src/main.ts

```diff
@@ -21,8 +22,13 @@ export async function activate(context: vscode.ExtensionContext) {
     const testHub = testExplorerExtension.exports;
 
     // this will register an ExUnitAdapter for each WorkspaceFolder
+    const exUnitRunner = new ExUnitRunner(workspaceFolder.name, workspaceFolder.uri.fsPath);
     context.subscriptions.push(
-      new TestAdapterRegistrar(testHub, (workspaceFolder) => new ExUnitAdapter(workspaceFolder, log), log)
+      new TestAdapterRegistrar(
+        testHub,
+        (workspaceFolder) => new ExUnitTestAdapter(exUnitRunner, workspaceFolder, log),
+        log
+      )
```

Comment:
There is a bug here. Line `25` uses first workspaceFolder to create a runner, where it should use `workspaceFolder` from line `29`. I think you could initialize exUnitRunner inside ExUnitTestsAdapter's constructor.